### PR TITLE
#95 Setup l10n in library

### LINKED ISSUE #95: Setup i18n in the Library
Make sure the i18n is available to be used from within the ODS Flutter Library

### PR DESCRIPTION
**Pull request by Tayebsed93**
Closes #95

### CODE REVIEW

**Comment by julien-deramond** [comment]
## Reminder to update the release process of the library

**Development process**

* Developer modify `ods_flutter_en.arb` and/or `ods_flutter_fr.arb`
* In order to have the translations in the app launched locally, `flutter gen-l10n` must be launched in the `lib` directory
  * It will regenerate:
    * `ods_flutter_localizations.dart`
    * `ods_flutter_localizations_en.dart`
    * `ods_flutter_localizations_fr.dart` 
 * The 5 files need to be committed and pushed

**Release the library**
* Pre-step: run `flutter gen-l10n` for security before creating and publishing the package
  * If files are modified, it means that something is wrong (`ods_flutter_localizations*` are maybe not all pushed)

**Review by julien-deramond — CHANGES_REQUESTED** [review]
Neat!

Only comment on my side: IMO `ods_flutter_localization*` files shouldn't be pushed because generated automatically by `flutter gen-l10n` and added in our `.gitignore`.
It will be clearer that the only files to modify are the `ods_flutter*.arb` and will avoid mistakes in general.

**Review by julien-deramond — APPROVED** [review]

**Comment by julien-deramond** [comment]
@Tayebsed93 I didn't see it at first but the tests are broken. Might be something to update to make them work again.

**Review by julien-deramond — CHANGES_REQUESTED** [review]
Based on the internal discussion:

- Lib must contains only wordings used within the lib (we can create a fake one for test purpose, but right now should be rather empty)
- The generation of the files should be done only in the app
- App should have the keys to translate for its own wording
- App should be able to translate the elements in the lib (+ its own) in another language

File: .github/workflows/demo_app.yml [hunk 1]
```diff
@@ -11,6 +11,10 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: subosito/flutter-action@v1
+      - name: Generate internationalization files
+        run: |
+          cd lib
```
Comment (ludovic35) [inline .github/workflows/demo_app.yml:16]: rename lib to library

File: app/lib/main.dart [hunk 2]
```diff
@@ -45,9 +45,10 @@ class OdsApplication extends StatelessWidget {
           debugShowCheckedModeBanner: false,
           home: MainScreen(),
           // Localization setup
-          supportedLocales: L10n.all,
+          supportedLocales: AppLocalizations.supportedLocales,
           localizationsDelegates: [
             AppLocalizations.delegate,
+            OdsLocalizations.delegate,
```
Comment (ludovic35) [inline app/lib/main.dart:51]: See if it is mandatory
If yes add a section in ReadMe how to Use

File: lib/lib/l10n/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+gen/
```
Comment (ludovic35) [inline lib/lib/l10n/.gitignore:1]: Remove the gitignore 
and add it at the top level

File: app/lib/main.dart (hunk 2 above)
Comment (ludovic35) [reply app/lib/main.dart:51]: Done

File: lib/lib/l10n/.gitignore (hunk 3 above)
Comment (ludovic35) [reply lib/lib/l10n/.gitignore]: Done

**Pull request merged**